.github/workflows/ci.yml

```diff
@@ -53,6 +53,46 @@ jobs:
           docker push ${{ env.DOCKER_REGISTRY_URL }}/ucentralgw:$tag
         done
 
+  docker-compose:
+    runs-on: ubuntu-20.04
+    env:
+      DOCKER_COMPOSE_PROJECT_NAME: ucentral-${{ github.sha }}
+    needs: docker
+    if: github.event_name == 'pull_request'
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+
+    - name: Create certificate and key files from Github secrets
+      env:
+        DIGICERT_TIP_WILDCARD_CERT: ${{ secrets.DIGICERT_TIP_WILDCARD_CERT }}
+        DIGICERT_TIP_WILDCARD_KEY: ${{ secrets.DIGICERT_TIP_WILDCARD_KEY }}
+      run: |
+        cp certificates/root.pem docker-compose/ucentral-data/certs/restapi-ca.pem
+        echo "$DIGICERT_TIP_WILDCARD_CERT" > docker-compose/ucentral-data/certs/websocket-cert.pem
+        echo "$DIGICERT_TIP_WILDCARD_CERT" > docker-compose/ucentral-data/certs/restapi-cert.pem
+        echo "$DIGICERT_TIP_WILDCARD_KEY" > docker-compose/ucentral-data/certs/websocket-key.pem
+        echo "$DIGICERT_TIP_WILDCARD_KEY" > docker-compose/ucentral-data/certs/restapi-key.pem
+
+    - name: Instantiate Docker Compose deployment
+      env:
+        UCENTRALGW_TAG: ${{ github.sha }}
+      run: |
+        docker-compose -f docker-compose/docker-compose.yml --project-directory docker-compose/ up -d
+        sleep 10
```

Comment:
@oblom0v just to add my 2cents - I usually try to avoid sleep. Can we try something like https://github.com/ufoscout/docker-compose-wait if that's relatively easy to work with?

Reply:
@leonidlm problem is that this solution only checks if the TCP port is open, which doesn't necessarily mean that the application is ready to process requests. Another solution would be to enable ALB support in the configuration, expose the port and insert a GitHub action which executes a curl request several times until it gets a successful health check response.

Reply:
@oblom0v sounds good, if we have a healthcheck endpoint already implemented in the application, let's use it to understand when the stack is ready for the next step, which is running the test.